index.js

```diff
@@ -1 +1,43 @@
-const app = "I don't do much."
+const products = [
+  { name: 'Head & Shoulders Shampoo', price: 5, discount: .6 },
+  { name: 'Twinkies', price: 8, discount: .45 },
+  { name: 'Oreos', price: 6.5, discount: .8 },
+  { name: 'Jasmine-scented bath pearls', price: 14, discount: .7 },
+];
+
+const couponLocations = [
+  { room: 'Living room', amount: 5 },
+  { room: 'Kitchen', amount: 2 },
+  { room: 'Bathroom', amount: 1 },
+  { room: 'Master bedroom', amount: 7 },
+];
+
+function reduce(collection, callback, initialValue) {
+  let result = initialValue;
+
+  collection.forEach(function (product) {
+    result = callback(result, product);
+  });
+
+  return result;
+}
+
+function callback(totalPrice, product) {
```

Comment:
This is a really minor note that, since we're shadowing this function (it'll get hoisted to the top and be shadowed by the `callback` arg in `reduce`), it _might_ be confusing for students.

I don't think we need to change anything -- the scope is clear -- but thought I'd leave a note just the same.